faq/font_sample.php

```diff
@@ -47,35 +50,62 @@
 // print page header
 echo "<h1>$title</h1>\n";
 
-echo "<p>" . _("This page demonstrates the differences between <span style='font-family: DPCustomMono2'>DPCustomMono2</span> and other fonts.") . "</p>\n";
+echo "<p>" . _("This page shows samples of the available proofreading fonts and demonstrates the differences between DPCustomMono2 and other fonts.") . "</p>\n";
+
+echo "<h2>" . _("Available Proofreading Fonts") . "</h2>";
+
+echo "<p>" . sprintf(_("The following fonts can be selected in your <a href='%s'>preferences</a> for use in the proofreading interface. Browser default is whatever font your browser renders monospace text in unless told otherwise, often Courier or Courier New. The other fonts are available as web fonts and can be selected and used without having them installed on your computer."), "$code_url/userprefs.php?tab=1") . "</p>";
+
+$show_user_custom_font = TRUE;
+foreach(get_available_proofreading_font_faces() as $index => $name)
+{
+    if($index == 1) // other
+        continue;
+
+    if($index == 0)
+    {
+        $name = BROWSER_DEFAULT_STR;
+        $font = 'monospace';
+    }
+    else
+    {
+        $font = $name;
+    }
 
-echo "<p>" . sprintf(_("DPCustomMono2 is a font adapted by DP volunteers, based on the suggestions and ideas of many experienced proofreaders. Using DPCustomMono2 as your proofreading font can help you find mistakes. You can change the font that you use for proofreading in your <a href='%s'>preferences</a>."), "$code_url/userprefs.php") . "</p>\n";
+    if($font == $proofreading_font)
+        $show_user_custom_font = FALSE;
 
-if(!empty($proofreading_font))
-    echo "<p>" . sprintf(_("Your current proofreading font is <b>%s</b>."), $proofreading_font) . "</p>\n";
+    show_font_specimen($name, $font, $proofreading_font);
+}
 
-echo "<p>" . _("This site provides DPCustomMono2 as a web font and browsers that support it do not require the font to be installed locally.") . "</p>";
+if($show_user_custom_font)
+{
+    echo "<h2 style='clear: both;'>" . _("Custom Proofreading Font") . "</h2>";
+    echo "<p>" . _("Your current proofreading font is one you've specified by name. This is what a specimen looks like in that font.") . "</p>";
+    show_font_specimen($proofreading_font, $proofreading_font);
+}
 
-echo "<p style='font-family: DPCustomMono2'>" . _("If your browser supports web fonts or you already have DPCustomMono2 installed, you will see this paragraph in that typeface. If this paragraph's font doesn't look radically different to that of the paragraph above, you can download DPCustomMono2 from <a href='DPCustomMono2.ttf'>here</a> (right click the link, and choose Save Target As..) After you have installed the font please refresh this page to make sure DPCustomMono2 is installed correctly.") . "</p>\n";
+echo "<h2 style='clear: both;'>" . _("DPCustomMono2") . "</h2>";
```

Comment:
Not sure we should be localizing font names, so I thought I'd ask. (I realize that it's embedded in other strings that do need to be localized, though.)

Reply:
No, we shouldn't - good catch!